### ISSUE TITLE
Sign-in and Sign-up design

### ISSUE THREAD

**Issue by stursberg** [hpi-swt2/sport-portal#185]
Design von Seiten "Anmelden" und "Registrieren"
- all data (for example font size, colors etc - in mock, ask Dima for CSS-Code);
- FONT: ProximaNova
- footer is implemented separately in #186 !

**Estimation: XL**

![Screen Shot 2017-11-29 at 17.55.54.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/59fc457019d4e4001305101f/125516c66e82c728ace21e0d46db978826878de9bce1b057e40fa0c57e0b76226456fa6b7191a379e80c2e54415a4abe5d4a1f06a3a3dc78e2a436628f0d0beed23f0b26933a43a9b8841fbf02b325df1e87e7ba9850d2aa6c7bb3d4d216744b470a83ed5c3daa984ec959ff4ce54a54a11b376bf43b.png)
 
 
Placeholder shouldn't disappear but move upwards (if not doable, ask Dima for new design without transitions), remove link 'Registrieren'
![Screen Shot 2017-11-29 at 17.56.05.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/59fc457019d4e4001305101f/125516c66e82c728ace21e0d46db978826878de9bce1b057e40fa0c57e0b76226456fa6b7191a379e80c2e57415f4bbe5d4a1f06a3a3dc78e2a436628f0d0beed23f0b26933a43a9b8841fbf02b325df1e87e7ba9850d2aa6c7bb3d4d216744b470a83ed5c3daa984ec959ff4ce54a56a01a3162f43b.png)
 
 
Both should be mobile-friendly: (navbar has background image)
![sms_11.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/59fc457019d4e4001305101f/125516c66e82c728ace21e0d46fb99891cd3d2ebc3abaf51ab57f5973d047d317c05f3782387e72aba1579545f594eac42515a10adfbd82db4aa2e32db025fb28c341521842f1bbeb98b15be04bb21824881b1b5824dc4ee2f68.png)
 

![sms_12.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/59fc457019d4e4001305101f/125516c66e82c728ace21e0d46fb99891cd3d1ebc3abaf51ab57f5973d047d317c05f3782387e72aba1579545f594eac42515a10adfbd82db4aa2e32db025fb28c341521842f1bbeba8215be04bb21824881bebd8d4bc2ed2f68.png)

- [x] Research on library for transitions on sign up page
- [x] Registrieren page
- [x] Anmelden page

**Comment by MarcelSchmidberger** [comment]
The mobile version of "Registrieren" has the field mail twice. @zhamanakov the first should probably be "Vorname" ?

**Comment by stursberg** [comment]
[ProximaNova.zip](https://github.com/hpi-swt2/sport-portal/files/1515240/ProximaNova.zip)
[sign-in-up.zip](https://github.com/hpi-swt2/sport-portal/files/1515241/sign-in-up.zip)

**Comment by jkbe** [comment]
http://materializecss.com/ might help. Please specify subtasks if that helps you.

**Comment by jcremerius** [comment]
Created branch flexibus/#185/Sign-in_and_Sign-up_design. Please pull and use this one for this issue :)

**Comment by jkbe** [comment]
@jcremerius can you please **always** reference the issue in your commit message using #<issue-number> at the end of the headline? Then it will appear in this thread here automatically and there's no need to reference it manually like this.

Before: Remove the obsolete registration reference on the sign_up page
After: Remove the obsolete registration reference on the sign_up page #185

**Comment by deeps96** [comment]
[To discuss] Remove Twitter-Bootstrap if all layout features have been implemented as we shouldn´t need this any longer?

**Comment by deeps96** [comment]
[Interesting Links]
- http://materializecss.com/
- http://guides.rubyonrails.org/form_helpers.html

**Comment by deeps96** [comment]
@zhamanakov Refering to the movile version of the sign in page: Should the register element really be a button? - In the desktop version it´s just a link. Just seems a bit inconsitent, but it can get implemented without any problems, if you wish.

**Comment by deeps96** [comment]
TODO´s for Sign-in page

- [x] remove background image on mobile devices
- [x] add blue border for focused text fields

**Comment by jkbe** [comment]
@zhamanakov checks license of background image, please don't merge before license is verified. HPI pictures to come.

**Comment by stursberg** [comment]
neues Background1 (nicht web-optimiert!)
![background_1](https://user-images.githubusercontent.com/15203334/33734461-b3e7ba3a-db8c-11e7-97e8-1b1e86b5040f.png)

**Comment by jcremerius** [comment]
Are there already links wich give us access to the pictures or de we have to set up our own image server?
Thanks!
edit: clicking on the picture enlightened me!

**Comment by jcremerius** [comment]
How to overwrite twitter bootstraps css of the navbar: https://stackoverflow.com/questions/18529274/change-navbar-color-in-twitter-bootstrap-3

**Comment by MariusDanner** [comment]
When this branch is checked out, other sites (like the events page) won't render properly. Is this an issue with this branch?
I also noticed that the text color changes on the sign in page when you are using a width of under 600 in mobile mode.

**Comment by jonashering** [comment]
Apparently because of the line `*= require_tree` in `application.scss`, all css classes (also those that are intended to be scoped to certain views) seem to become globally available. We should be able to fix that. Thanks for the comment!

**Comment by jkbe** [comment]
_Ongoing User Story_

**Estimation for this Sprint: M (Actual: L)**

Further tasks:
- [x] Come up with a way to host images / assets on HPI server
- [x] Come up with a tidy way (no if / else if possible) to distinguish web from mobile for background image
- [x] Implement giving hints for invalid inputs (for now, leave the design for this as is standard, ask Dima if it's ok)
- [x] web-optimize background image as it's 2,5 MB in size at the moment and takes a while to load
- [x] fix broken font in Heroku deployment (see tips below)
- [x] https://github.com/hpi-swt2/sport-portal/issues/185#issuecomment-352268746

**Comment by jak-ing** [comment]
_Ongoing User Story_
**Estimation for this Sprint: M**

**Comment by jkbe** [comment]
@MariusDanner @jcremerius you could investigate on sport-portal.herokuapp.com why the new font type is not properly displayed when deployed to Heroku.

**Comment by jkbe** [comment]
I just created a new branch `flexibus/#185-2/Sign-in-and-Sign-up-design` that contains the revert of the revert PR + your three recent commits so that in the resulting PR, we have a better overview of all the code created to solve User Story #185 (and not just the three most recent commits...).
Please continue work on `flexibus/#185-2/Sign-in-and-Sign-up-design`, I will delete the old branch `flexibus/#185/Sign-in_and_Sign-up_design` shortly (edit: deleted).

**Comment by jkbe** [comment]
In regard to this: https://github.com/hpi-swt2/sport-portal/commit/6850c04e6c3993146e2955838f9e956427391c33#diff-f859e2848b07324f808ce1a5ccd9fcbaR21
I'd suggest to change the font family definition for our custom font to something like this: https://github.com/hpi-swt2/sport-portal/commit/f3216c31150be3d6b2f09b4ebe393c3e68bb6850#diff-f859e2848b07324f808ce1a5ccd9fcba (on my branch)
This way, `<b>` and `<strong>` elements automatically pick up on the bold version of Proxima Nova and we don't have to set `font-type` manually on all these elements --> less & cleaner code.

**Comment by jkbe** [comment]
Oh and I'd guess that the broken Proxima Nova font on current `master` has something to do with how assets are bundled differently depending on development/production environment (see `config/environments`). @chrisma do you have a tip for a solution for us here? The new font was added to `assets/fonts` directory.

**Comment by chrisma** [comment]
Have you tried precompiling the assets (http://guides.rubyonrails.org/asset_pipeline.html#precompiling-assets)?
`RAILS_ENV=production bin/rails assets:precompile`

**Comment by deeps96** [comment]
What's the current status of this user story? Since #182 depends on materialize and reusable componentens introduced by this story: Whats the estimated time required for completion?

**Comment by jcremerius** [comment]
What we have not done yet is the heroku and the invalid input issue. As we have not heard anything from @zhamanakov, I suggest to leave it as it is, so you can start working. The heroku thing is completely independent of the code, so we could create a new issue to handle this. Would that be an option @jak-ing? If that is ok we can create the pull request immediately.
The pictures are hosted on my owncloud. I can share it with everyone who wants to upload pictures. @jak-ing maybe you can send me a list of users who need permission to upload pictures? Thank you!

**Comment by jkbe** [comment]
@jcremerius I also talked to @MariusDanner today and he said he would look into the remaining work for this issue. As for the broken Heroku deployment: We could probably "test-deploy" the PR of this issue before merging so that we can test whether the problem with importing assets is solved now, I'll look into that. Let's try to make sure that the font is displayed correctly before merging into dev.
Regarding the owncloud: It would be good if you could add everybody from Team Flexibus to the folder. I'll write a message in Slack so that other people can contact us if they need to add pictures etc.
@deeps96 if you want to start work on #182, you could also create a new branch for that which is based off the branch of this issue (I can assist with that if need be) so that you don't have to wait until 185 is merged.

**Comment by jcremerius** [comment]
Sounds good. Everyone in our team is now able to use the owncloud. Thank you! :)

**Comment by jkbe** [comment]
Branch `flexibus/#185-2/Sign-in-and-Sign-up-design` is now automatically deployed on each commit (if Travis, Coveralls passed) to http://sport-portal-pr.herokuapp.com/. So you should be able to resolve the font-bug before merging @MariusDanner @jcremerius. ;)

**Comment by jcremerius** [comment]
![error_mob](https://user-images.githubusercontent.com/15142695/34709289-efa51d44-f516-11e7-9a52-a5791ad40942.png)
![error_desk](https://user-images.githubusercontent.com/15142695/34709290-efc44bec-f516-11e7-9cb5-d1a5948fc5be.png)
